configure

```diff
@@ -804,6 +805,25 @@ if enabled_or_auto libsystemd_daemon; then
   fi
 fi
 
+#
+# year2038
+#
+check_cc_snippet year2038 '
+#define _FILE_OFFSET_BITS 64
+#define _TIME_BITS 64
+#include <time.h>
+int test(void){
+  #define LARGE_TIME_T ((time_t) (((time_t) 1 << 30) - 1 + 3 * ((time_t) 1 << 30)))
+  int verify_time_t_range[(LARGE_TIME_T / 65537 == 65535 && LARGE_TIME_T % 65537 == 0) ? 1 : -1];
+  return 0;
+}
+' -Werror
+
+if disabled year2038; then
+  CFLAGS="-D_USE_32BIT_TIME_T ${CFLAGS}"
```

Comment:
I don't think you are supposed to set a cflag, as `CONFIG_YEAR238` and `ENABLE_YEAR238` now exist in build.h